settings.py

```diff
@@ -270,11 +270,15 @@ def dump(self, f: TextIO, level: int = 0) -> None:
             # fetch class to avoid going through getattr
             cls = self.__class__
             type_hints = cls.get_type_hints()
+            entries = [e for e in self]
+            if not type_hints and not entries:
```

Comment:
Sorry for the late response.

I think we can ignore type_hints here. The actual dumping only uses type_hints for lookup (and assert for world devs to find lines where they forgot to assign a default value), so checking it there seems a bit confusing.
```suggestion
            if not entries:
```